types/sarif/index.d.ts

```diff
@@ -0,0 +1,1607 @@
+     */
+    codeFlows?: CodeFlow[];
+
+    /**
+     * An array of physicalLocation objects which specify the portions of an analysis tool's output that a converter
+     * transformed into the result object.
+     */
+    conversionProvenance?: PhysicalLocation[];
+
+    /**
+     * A stable, unique identifier for the equivalence class of logically identical results to which this result
+     * belongs, in the form of a GUID.
+     */
+    correlationGuid?: string;
+
+    /**
+     * A set of strings each of which individually defines a stable, unique identity for the result.
+     */
+    fingerprints?: { [key: string]: string };
+
+    /**
+     * An array of 'fix' objects, each of which represents a proposed fix to the problem indicated by the result.
+     */
+    fixes?: Fix[];
+
+    /**
+     * A dictionary, each of whose keys is the id of a graph and each of whose values is a 'graph' object with that
+     * id.
+     */
+    graphs?: { [key: string]: Graph };
+
+    /**
+     * An array of one or more unique 'graphTraversal' objects.
+     */
+    graphTraversals?: GraphTraversal[];
+
+    /**
+     * A stable, unique identifer for the result in the form of a GUID.
+     */
+    instanceGuid?: string;
+
+    /**
+     * A value specifying the severity level of the result.
+     */
+    level?: Result.level;
+
+    /**
+     * One or more locations where the result occurred. Specify only one location unless the problem indicated by
+     * the result can only be corrected by making a change at every specified location.
+     */
+    locations?: Location[];
+
+    /**
+     * A message that describes the result. The first sentence of the message only will be displayed when visible
+     * space is limited.
+     */
+    message?: Message;
+
+    /**
+     * A positive integer specifying the number of times this logically unique result was observed in this run.
+     */
+    occurrenceCount?: number;
+
+    /**
+     * A set of strings that contribute to the stable, unique identity of the result.
+     */
+    partialFingerprints?: { [key: string]: string };
+
+    /**
+     * A set of locations relevant to this result.
+     */
+    relatedLocations?: Location[];
+
+    /**
+     * The stable, unique identifier of the rule (if any) to which this notification is relevant. This member can be
+     * used to retrieve rule metadata from the rules dictionary, if it exists.
+     */
+    ruleId?: string;
+
+    /**
+     * An array of 'stack' objects relevant to the result.
+     */
+    stacks?: Stack[];
+
+    /**
+     * A set of flags indicating one or more suppression conditions.
+     */
+    suppressionStates?: Result.suppressionStates[];
+
+    /**
+     * The URIs of the work items associated with this result
+     */
+    workItemUris?: string[];
+
+    /**
+     * Key/value pairs that provide additional information about the result.
+     */
+    properties?: PropertyBag;
+}
+
+export namespace Result {
```

Comment:
See previous comment.

Reply:
Thanks for the feedback, I've updated these to be "type" instead of "const enum"